numpy/lib/arraypad.py

```diff
@@ -269,17 +209,31 @@ def _get_linear_ramps(padded, axis, width_pair, end_value_pair):
     """
     edge_pair = _get_edges(padded, axis, width_pair)
 
-    left_ramp = _linear_ramp(
-        padded.ndim, axis, start=end_value_pair[0], stop=edge_pair[0],
-        size=width_pair[0], reverse=False
+    left_ramp = np.linspace(
+        start=end_value_pair[0],
+        stop=edge_pair[0],
+        num=width_pair[0],
+        endpoint=False,
+        dtype=padded.dtype,
+        axis=axis,
     )
-    _round_if_needed(left_ramp, padded.dtype)
-
-    right_ramp = _linear_ramp(
-        padded.ndim, axis, start=end_value_pair[1], stop=edge_pair[1],
-        size=width_pair[1], reverse=True
+    if left_ramp.ndim > 1:
+        # linspace inserts new axis, remove old one with size 1
+        left_ramp = np.squeeze(left_ramp, axis + 1)
+
+    right_ramp = np.linspace(
+        start=end_value_pair[1],
+        stop=edge_pair[1],
+        num=width_pair[1],
+        endpoint=False,
+        dtype=padded.dtype,
+        axis=axis,
     )
-    _round_if_needed(right_ramp, padded.dtype)
+    if right_ramp.ndim > 1:
+        # linspace inserts new axis, remove old one with size 1
+        right_ramp = np.squeeze(right_ramp, axis + 1)
```

Comment:
Looks good to me, I would not worry about the speed, at least for now. I do not quite understand why the `if` is needed above, as far as I can tell the linspace output must be 2D? (you may also just use `right_ramp.squeeze(axis + 1)`)

Reply:
Good catch about the unnecessary if-statement!